vars/DevelopPipeline.groovy

```diff
@@ -1,81 +1,94 @@
-                LTD_PASSWORD="${user_ci_PSW}"
-                work_branches = "${GIT_BRANCH} ${CHANGE_BRANCH} develop"
-                idl_name = ${idl_name}
-                XML_REPORT= "jenkinsReport/report.xml"
-                MODULE_NAME = ${module_name}
+    Csc csc = new Csc()
+    properties(
+        [
+        buildDiscarder(
+            logRotator(
+                artifactDaysToKeepStr: '',
+                artifactNumToKeepStr: '',
+                daysToKeepStr: '14',
+                numToKeepStr: '10',
+            )
+        ),
+        // Make new builds terminate existing builds
+        disableConcurrentBuilds(
+            abortPrevious: true,
+        )
+        ]
+    )
+    pipeline {
+        agent {
+            docker {
+                alwaysPull true
+                image 'lsstts/develop-env:develop'
+                args "--entrypoint=''"
             }
+        }
+        environment {
+            user_ci = credentials('lsst-io')
+            LTD_USERNAME="${user_ci_USR}"
+            LTD_PASSWORD="${user_ci_PSW}"
+            WORK_BRANCHES = "${GIT_BRANCH} ${CHANGE_BRANCH} develop"
+            IDL_NAME = "${idl_name}"
+            XML_REPORT= "jenkinsReport/report.xml"
+            MODULE_NAME = "${module_name}"
+        }
 
-            stages {
-                stage ('Install Requirements and Update Branches') {
-                    steps {
-                        withEnv(["HOME=${env.WORKSPACE}"]) {
-                            sh """
-                                source /home/saluser/.setup_dev.sh || echo loading env failed. Continuing...
-                                cd /home/saluser/repos/ts_xml
-                                /home/saluser/.checkout_repo.sh ${work_branches}
-                                git pull
-                                cd /home/saluser/repos/ts_salobj
-                                /home/saluser/.checkout_repo.sh ${work_branches}
-                                git pull
-                                cd /home/saluser/repos/ts_sal
-                                /home/saluser/.checkout_repo.sh ${work_branches}
-                                git pull
-                                cd /home/saluser/repos/ts_idl
-                                /home/saluser/.checkout_repo.sh ${work_branches}
-                                git pull
-                                make_idl_files.py ${idl_name}
-                            """
+        stages {
+            stage ('Install Requirements and Update Branches') {
+                steps {
+                    withEnv(["WHOME=${env.WORKSPACE}"]) {
+                        script {
+                            csc.update_branches("ts_xml")
+                            csc.update_branches("ts_salobj")
+                            csc.update_branches("ts_sal")
+                            csc.update_branches("ts_idl")
+                            csc.make_idl_files("${IDL_NAME}")
```

Comment:
Some components interact with other systems and will need to build the IDL files for them. There is also the case where packages need all IDLs, and use `--all` instead. I think the easiest way to support this is to have a separate parameter for the IDLs to build, e.g.; `required_idl`. We could be fancy about it and do things like if `required_idl` is not provided use `idl_name` and if `required_idl` is not `--all` concatenate `idl_name` with `required_idl`. Not sure how hard that will be with groovy though.